Rubberduck.Parsing/Annotations/VBAParserAnnotationFactory.cs

```diff
@@ -8,61 +8,29 @@ namespace Rubberduck.Parsing.Annotations
-            _creators.Add(AnnotationType.Description.ToString().ToUpperInvariant(), typeof (DescriptionAnnotation));
-            _creators.Add(AnnotationType.PredeclaredId.ToString().ToUpperInvariant(), typeof(PredeclaredIdAnnotation));
-            _creators.Add(AnnotationType.DefaultMember.ToString().ToUpperInvariant(), typeof(DefaultMemberAnnotation));
-            _creators.Add(AnnotationType.Enumerator.ToString().ToUpperInvariant(), typeof(EnumeratorMemberAnnotation));
-            _creators.Add(AnnotationType.Exposed.ToString().ToUpperInvariant(), typeof (ExposedModuleAnnotation));
-            _creators.Add(AnnotationType.Obsolete.ToString().ToUpperInvariant(), typeof(ObsoleteAnnotation));
-            _creators.Add(AnnotationType.ModuleAttribute.ToString().ToUpperInvariant(), typeof(ModuleAttributeAnnotation));
-            _creators.Add(AnnotationType.MemberAttribute.ToString().ToUpperInvariant(), typeof(MemberAttributeAnnotation));
-            _creators.Add(AnnotationType.ModuleDescription.ToString().ToUpperInvariant(), typeof(ModuleDescriptionAnnotation));
-            _creators.Add(AnnotationType.ExcelHotKey.ToString().ToUpperInvariant(), typeof(ExcelHotKeyAnnotation));
-        }
-
-        public IAnnotation Create(VBAParser.AnnotationContext context, QualifiedSelection qualifiedSelection)
-        {
-            var annotationName = context.annotationName().GetText();
-            var parameters = AnnotationParametersFromContext(context);
-            return CreateAnnotation(annotationName, parameters, qualifiedSelection, context);
-        }
-
-            private static List<string> AnnotationParametersFromContext(VBAParser.AnnotationContext context)
+            foreach (var annotation in recognizedAnnotations)
             {
-                var parameters = new List<string>();
-                var argList = context.annotationArgList();
-                if (argList != null)
+                if (annotation is NotRecognizedAnnotation)
                 {
-                    parameters.AddRange(argList.annotationArg().Select(arg => arg.GetText()));
+                    unrecognized = annotation;
                 }
-                return parameters;
+                _lookup.Add(annotation.Name.ToUpperInvariant(), annotation);
             }
+        }
 
-        private IAnnotation CreateAnnotation(string annotationName, IReadOnlyList<string> parameters,
-            QualifiedSelection qualifiedSelection, VBAParser.AnnotationContext context)
+        public ParseTreeAnnotation Create(VBAParser.AnnotationContext context, QualifiedSelection qualifiedSelection)
         {
-            if (_creators.TryGetValue(annotationName.ToUpperInvariant(), out var annotationClrType))
+            var annotationName = context.annotationName().GetText();
+            if (_lookup.TryGetValue(annotationName.ToUpperInvariant(), out var annotation))
```

Comment:
Nitpick: everywhere else where we don't care about lettercases, we use `ToLowerInvariant()`. Obviously it doesn't matter but consistency?

Reply:
so be it :)